consensus/pbft/viewchange.go

```diff
@@ -435,23 +435,23 @@ func (instance *pbftCore) processNewView() events.Event {
 				continue
 			}
 
-			if _, ok := instance.reqStore[d]; !ok {
-				logger.Warningf("Replica %d missing assigned, non-checkpointed request %s",
+			if _, ok := instance.reqBatchStore[d]; !ok {
+				logger.Warningf("Replica %d missing assigned, non-checkpointed request batch %s",
 					instance.id, d)
-				if _, ok := instance.missingReqs[d]; !ok {
-					logger.Warningf("Replica %v requesting to fetch %s",
+				if _, ok := instance.missingReqBatches[d]; !ok {
+					logger.Warningf("Replica %v requesting to fetch batch %s",
 						instance.id, d)
-					newRequestMissing = true
-					instance.missingReqs[d] = true
+					newReqBatchMissing = true
+					instance.missingReqBatches[d] = true
 				}
 			}
 		}
 	}
 
-	if len(instance.missingReqs) == 0 {
+	if len(instance.missingReqBatches) == 0 {
 		return instance.processNewView2(nv)
-	} else if newRequestMissing {
-		instance.fetchRequests()
+	} else if newReqBatchMissing {
+		instance.fetchRequestBatches()
```

Comment:
I understand this is not a new mechanism, but since it's been touched in this changeset, I think it's at least worth some discussion.  I'm not sure why we generally expect that we can fetch missing requests from other replicas.  In general, I would expect that if other replicas already have the requests, then they will quickly execute to a checkpoint and garbage collect the old requests.

In many ways, it almost seems like the new view message should include the requests which are being reprocessed in the new view.  It would balloon the new view message, but, view change should be a rare case, and, with periodic view change enabled, things are lined up on checkpoint boundaries, expressly to avoid having large xSet/pSet/qSets or having to fetch requests.